### PR TITLE
Use multi stage dockerfile

### PR DESCRIPTION
**Pull request by Sispheor**

### CODE REVIEW

**Comment by Sispheor** [comment]
This allow to pass from 1.4Gb to 800 MB.

**Comment by Sispheor** [comment]
Test are also ok when executed inside the image.
We need to test manually the LDAP auth as it's not yet integrated to our tests

File: docker/squest.Dockerfile [hunk 1]
```diff
@@ -32,51 +37,64 @@ RUN set -ex \
...
-ENV NODE_VERSION=12.22.6
-ENV NVM_VERSION=v0.38.0
 RUN curl -o- https://raw.githubusercontent.com/creationix/nvm/${NVM_VERSION}/install.sh | bash
-ENV NVM_DIR=/root/.nvm
 RUN . "$NVM_DIR/nvm.sh" && nvm install ${NODE_VERSION}
 RUN . "$NVM_DIR/nvm.sh" && nvm use ${NODE_VERSION}
 RUN . "$NVM_DIR/nvm.sh" && nvm alias default v${NODE_VERSION}
-ENV PATH="/root/.nvm/versions/node/v${NODE_VERSION}/bin/:${PATH}"
 RUN node --version
 RUN npm --version
```
Comment (a-belhadj) [inline docker/squest.Dockerfile:49]: You can delete it

File: docker/squest.Dockerfile [hunk 2]
```diff
@@ -32,51 +37,64 @@ RUN set -ex \
...
+    && RUN_DEPS=" \
+    default-mysql-client \
+    graphviz \
+    " \
+    && apt-get update && apt-get install -y --no-install-recommends $RUN_DEPS \
+    && rm -rf /var/lib/apt/lists/*
+
+# Create media and backup folders
+RUN mkdir -p /app/media && \
+    mkdir /app/backup
```
Comment (a-belhadj) [inline docker/squest.Dockerfile:85]: RUN mkdir -p /app/media /app/backup

**Review by a-belhadj — APPROVED** [review]

**Comment by Sispheor** [comment]
LDAP not ok. System packages are required.
I'll moved them in the python base stage.

File: docker/squest.Dockerfile (hunk 1 above)
Comment (Sispheor) [reply docker/squest.Dockerfile]: done

File: docker/squest.Dockerfile (hunk 2 above)
Comment (Sispheor) [reply docker/squest.Dockerfile]: done

**Comment by Sispheor** [comment]
LDAP tested and ok.
Final image size 1GB. So total saved space almost 500 MB.

It's the best I can do. Some python libs require binaries that comes with system packages.
I could have copied only the lib but it would be painful to maintain in time just to save 200 MB of storage.

@a-belhadj what do you think?

**Comment by a-belhadj** [comment]
> LDAP tested and ok.
> Final image size 1GB. So total saved space almost 500 MB.
> 
> It's the best I can do. Some python libs require binaries that comes with system packages.
> I could have copied only the lib but it would be painful to maintain in time just to save 200 MB of storage.
> 
> @a-belhadj what do you think?

It looks good to me 
We can reduce another 100MB by switching to an alpine base (6MB vs 120MB) but it's not worth it, we'd have to redo the entire dockerfile

**Review by a-belhadj — APPROVED** [review]

**Pull request merged**